internal/store/postgres/postgres.go

```diff
@@ -19,8 +19,13 @@ var (
 	errForeignKeyViolation = errors.New("foreign key violation")
 )
 
-// New returns the database instance
-func New(logger log.Logger, c Config) (*gorm.DB, error) {
+type Client struct {
```

Comment:
this adds a new abstraction for the store, any reason to have a client struct here?

Reply:
To decouple store from implementation specific e.g. gorm vs sqlx